proposals/type-aliases.md

```diff
@@ -131,13 +159,46 @@ typealias Encoded<E> = ByteArray         // Warning: generic type parameter E is
 >
 > fun processIsoEncoded(data: Encoded<Iso>) {
 >   // Both 'Encoded<Iso>' and 'Encoded<Utf8>' are expanded to 'ByteArray'.
->   // So, regardless of the design intent, the following call is ok:
 >   processUtf8Encoded(data)
 > }
 >```
 >
 > However, if we prohibit such type aliases, it creates unnecessary long-term commitment for generic type aliases.
-> Since it is not an error for generic classes,
+> NB this is not an error for generic classes.
+
+Type arguments of a generic type alias should be well-formed types (type projections).
+```
+interface NN<T : Any>
+typealias Predicate<T> = (T) -> Boolean
+
+typealias E1 = Predicate<NN<Int, Int>>          // Error: wrong number of type arguments
+typealias E2 = Predicate<NN<Int?>>              // Error: upper bound violated
```

Comment:
Within the current concept, no additional rules are required: this will be reported as an error at the use site with detailed information on type alias expansion context).
Fixed.